nginx and nginx-cur build failures

The builds fail in the configuration phase as follows:

...
\+ ./configure --prefix=/sw --conf-path=/sw/etc/nginx/nginx.conf --error-log-path=/sw/var/log/nginx/error.log --http-client-body-temp-path=/sw/var/lib/nginx/body --http-fastcgi-temp-path=/sw/var/lib/nginx/fastcgi --http-log-path=/sw/var/log/nginx/access.log --http-proxy-temp-path=/sw/var/lib/nginx/proxy --http-scgi-temp-path=/sw/var/lib/nginx/scgi --http-uwsgi-temp-path=/sw/var/lib/nginx/uwsgi --lock-path=/sw/var/lock/nginx/nginx.lock --modules-path=/sw/var/lib/nginx/modules --pid-path=/sw/var/run/nginx.pid --with-pcre-jit --with-debug --with-http_addition_module --with-http_auth_request_module --with-http_dav_module --with-http_flv_module --with-http_geoip_module --with-http_gunzip_module --with-http_gzip_static_module --with-http_image_filter_module --with-http_mp4_module --with-http_perl_module --with-http_random_index_module --with-http_realip_module --with-http_secure_link_module --with-http_stub_status_module --with-http_ssl_module --with-http_sub_module --with-http_v2_module --with-http_xslt_module --with-ipv6 --with-mail --with-mail_ssl_module --with-perl=/usr/bin/perl --with-stream --with-stream_ssl_module --with-threads --user=www --group=www '--with-ld-opt=-Wl,-read_only_stubs -Wl,-bind_at_load -fPIE -Wl,-pie -L/sw/lib' '--with-cc-opt=-g -O2 -fstack-protector -Wformat -Werror=format-security -fPIE -D_FORTIFY_SOURCE=2 -I/sw/include' --with-perl_modules_path=/sw/lib/perl5/5.18.2
...
checking for libxslt ... not found
checking for libxslt in /usr/local/ ... not found
checking for libxslt in /usr/pkg/ ... not found
checking for libxslt in /opt/local/ ... not found

./configure: error: the HTTP XSLT module requires the libxml2/libxslt
libraries. You can either do not enable the module or install the libraries.

The actual detection command from %p/src/fink.build/nginx-1.12.2-1/nginx-1.12.2/objs/autoconf.err is:

checking for libxslt

objs/autotest.c:4:10: fatal error: 'libxml/parser.h' file not found
#include <libxml/parser.h>
         ^~~~~~~~~~~~~~~~~
1 error generated.
----------

#include <sys/types.h>
#include <unistd.h>
#include <libxml/parser.h>
                      #include <libxml/tree.h>
                      #include <libxslt/xslt.h>
                      #include <libxslt/xsltInternals.h>
                      #include <libxslt/transform.h>
                      #include <libxslt/xsltutils.h>

int main(void) {
    xmlParserCtxtPtr    ctxt = NULL;
                      xsltStylesheetPtr   sheet = NULL;
                      xmlDocPtr           doc;
                      doc = xmlParseChunk(ctxt, NULL, 0, 0);
                      xsltApplyStylesheet(sheet, doc, NULL);;
    return 0;
}

----------
cc -g -O2 -fstack-protector -Wformat -Werror=format-security -fPIE -g -O2 -fstack-protector -Wformat -Werror=format-security -fPIE -D_FORTIFY_SOURCE=2 -I/sw/include -I /usr/include/libxml2 -o objs/autotest objs/autotest.c -Wl,-read_only_stubs -Wl,-bind_at_load -fPIE -Wl,-pie -L/sw/lib -lxml2 -lxslt

Fink's libxml/parser.h is installed in a path that this command isn't including:

Fionna:Downloads hansen$ dpkg -S libxml/parser.h
libxml2: /sw/include/libxml2/libxml/parser.h

--- a/10.9-libcxx/stable/main/finkinfo/web/nginx.info
+++ b/10.9-libcxx/stable/main/finkinfo/web/nginx.info
@@ -1,5 +1,5 @@
 Package: nginx
-Version: 1.12.2
+Version: 1.14.2
 Revision: 1
 Description: Small, powerful, scalable web/proxy server
 License: BSD
@@ -15,8 +15,10 @@ Depends: <<
 	system-perl
 <<
 BuildDepends: <<
+	fink-package-precedence,
 	gd3,
 	geoip-dev,
+	libiconv-dev,
 	libpcre1,
 	libxml2,
 	libxslt,
@@ -28,15 +30,17 @@ Provides: httpd
 Replaces: nginx, nginx-cur
 Conflicts: nginx, nginx-cur
 Source: http://nginx.org/download/%n-%v.tar.gz
-Source-MD5: 4d2fc76211435f029271f1cf6d7eeae3
-Source-Checksum: SHA1(6b41d63befa4f52b0724b533e6292a6671b71fdc)
+Source-MD5: 239b829a13cea1d244c1044e830bd9c2
+Source-Checksum: SHA1(4b4df8786b44e79cffd2e002a070e27fd774a17f)
 PatchScript: <<
   /usr/bin/sed -i "" -e 's/"\$ENV{NGX_PM_CFLAGS}",/"\$ENV{NGX_PM_CFLAGS}", LDDLFLAGS => "\$Config{lddlflags} \$ENV{LDFLAGS}",/g' \
     -e 's/use 5.006001;/use 5.006001; use Config;/g' \
     src/http/modules/perl/Makefile.PL
+  # Don't try to detect MacPorts external libraries. Use Fink's.
+  perl -pi -e 's|/opt/local|%p|g; s|/usr/include|%p/include|g' auto/lib/{geoip,google-perftools,libgd,libxslt,openssl,pcre}/conf
 <<
 SetCFLAGS: -g -O2 -fstack-protector -Wformat -Werror=format-security -fPIE
-SetCPPFLAGS: -D_FORTIFY_SOURCE=2 
+SetCPPFLAGS: -D_FORTIFY_SOURCE=2 -MD
 SetCXXFLAGS: -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -fPIE
 SetLDFLAGS: -Wl,-read_only_stubs -Wl,-bind_at_load -fPIE -Wl,-pie
 ConfigureParams: <<
@@ -88,6 +92,7 @@ CompileScript: <<
  MYPERL="$(/usr/bin/perl -w -Mstrict -E 'my $vers = $^V; $vers =~ tr/[0-9.]//cd; print $vers . "\n";')"
  ./configure %c --with-perl_modules_path=%p/lib/perl5/$MYPERL
  make INSTALLSITEMAN3DIR=%p/share/man/man3
+ fink-package-precedence --depfile-ext='\.d' .
 <<
 InstallScript: <<
 #!/bin/sh -ex
@@ -182,6 +187,11 @@ DescDetail: <<
 <<
 DescPackaging: <<
  Former maintainer: Andreas "gecko" Gockel
+ 
+ Uses homebuilt library detection that assumes things in /usr/include. This misses libxml2
+ among others, so we make sure it finds our own:
+ https://github.com/fink/fink-distributions/issues/395
+ https://trac.nginx.org/nginx/ticket/1765
 <<
 InfoTest: <<
  TestScript: echo "Nothing to test here"

